src/java.base/share/classes/java/lang/StringConcatHelper.java

```diff
@@ -555,9 +555,6 @@ static byte[] newArray(long indexCoder) {
      */
     @ForceInline
     static byte[] newArray(int length) {
-        if (length < 0) {
-            throw new OutOfMemoryError("Overflow: String length out of range");
```

Comment:
I think we should keep this check, as string concatenation seems to use this to throw OOME.